src/python/tests/core/api/test_browser_assets.py

```diff
@@ -0,0 +1,635 @@
+        hours is passed as query parameter
+        http://localhost:8082/foglamp/asset/TESTAPI/x?hours=2
+        """
+        conn = http.client.HTTPConnection(BASE_URL)
+        conn.request("GET", '/foglamp/asset/{}/{}?hours={}'.format(test_data_asset_code, sensor_code_1, 2))
+        r = conn.getresponse()
+        assert 200 == r.status
+        r = r.read().decode()
+        conn.close()
+        retval = json.loads(r)
+        assert 3 == len(retval)
+        assert retval[0][sensor_code_1] == self.test_data_x_val_list[-1]
+        assert retval[0]['timestamp'] == self.test_data_ts_list[-1]
+        assert retval[1][sensor_code_1] == self.test_data_x_val_list[-2]
+        assert retval[1]['timestamp'] == self.test_data_ts_list[-2]
+        assert retval[2][sensor_code_1] == self.test_data_x_val_list[-3]
+        assert retval[2]['timestamp'] == self.test_data_ts_list[-3]
+
+    @pytest.mark.xfail(reason="FOGL-545")
+    async def test_get_asset_sensor_readings_q_time_complex(self):
+        """
+        Verify that if a combination of hrs, min, sec is used, shortest period will apply for sensor reading
+        http://localhost:8082/foglamp/asset/TESTAPI/x?hours=20&minutes=20&seconds=120&limit=20
+        """
+        conn = http.client.HTTPConnection(BASE_URL)
+        conn.request("GET", '/foglamp/asset/{}/{}?hours={}&minutes={}&seconds={}&limit={}'
+                     .format(test_data_asset_code, sensor_code_1, 20, 20, 120, 20))
+        r = conn.getresponse()
+        assert 200 == r.status
+        r = r.read().decode()
+        conn.close()
+        retval = json.loads(r)
+        assert 1 == len(retval)
+        assert retval[0][sensor_code_1] == self.test_data_x_val_list[-1]
+        assert retval[0]['timestamp'] == self.test_data_ts_list[-1]
+
+    @pytest.mark.xfail(reason="FOGL-546, FOGL-547")
+    async def test_get_asset_sensor_readings_stats(self):
+        """
+        Verify that if more than 20 readings for an assets sensor value,
+        summary of only 20 readings are returned as the default limit
+        http://localhost:8082/foglamp/asset/TESTAPI/x/summary
+        """
+        conn = http.client.HTTPConnection(BASE_URL)
+        conn.request("GET", '/foglamp/asset/{}/{}/summary'.format(test_data_asset_code, sensor_code_1))
+        r = conn.getresponse()
+        assert 200 == r.status
+        r = r.read().decode()
+        conn.close()
+        retval = json.loads(r)
+        assert 1 == len(retval)
+        assert retval[sensor_code_1]['min'] == min(self.test_data_x_val_list[1:-1])
+        assert retval[sensor_code_1]['average'] == \
+               sum(self.test_data_x_val_list[1:-1])/len(self.test_data_x_val_list[1:-1])
+        assert retval[sensor_code_1]['max'] == max(self.test_data_x_val_list[1:-1])
+
+    @pytest.mark.xfail(reason="FOGL-546, FOGL-548")
+    async def test_get_asset_sensor_readings_stats_q_limit(self):
```

Comment:
will summary for a specific sensor will have limit? either limit be applied or not. it will always be 1. with avg/ min and max and that will be for all

Reply:
Or we want to find last n readings summary for a sensor? CHECK

Reply:
Its stats summary for last n readings (limit=n)